src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,223 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertFormatQuery =
+                "INSERT INTO cars (model, manufacturer_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createStatement =
+                        connection.prepareStatement(insertFormatQuery,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createStatement.setString(1, car.getModel());
+            createStatement.setLong(2, car.getManufacturer().getId());
+            createStatement.executeUpdate();
+            ResultSet resultSet = createStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", e);
+        }
+        insertDriver(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getQuery =
+                "SELECT c.id AS car_id, mf.name, model,country, manufacturer_id "
+                        + "FROM cars c JOIN manufacturers mf "
+                        + "ON c.manufacturer_id = mf.id "
+                        + "WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getByIdStatement =
+                        connection.prepareStatement(getQuery)) {
+            getByIdStatement.setLong(1, id);
+            ResultSet resultSet = getByIdStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
```

Comment:
I'm not sure, but maybe you can move `car.setDrivers(getDriversForCar(id))` inside the method `getCar(resultSet)`, because if you've initiated creating a car object then it means that car is not `null`.
But it's up to you.

Reply:
OK, I was wrong )